lib/instances.js

```diff
@@ -0,0 +1,342 @@
+    }
+    throw new BedrockError(
+      'Duplicate issuer instance.',
+      'DuplicateError', {
+        public: true,
+        httpStatusCode: 409
+      }, e);
+  }
+  // do not expose `controllerKeySeed`
+  delete record.controllerKeySeed;
+
+  // initialize instance
+  record.instance.keys = await _initialize({id: instance.id});
+
+  // return record
+  return record;
+};
+
+// get a particular instance
+exports.get = async ({id, controller}) => {
+  const query = {'instance.id': id};
+  // TODO: determine `public` status of information in instance
+  if(controller) {
+    // only allow find if controller matches
+    query.controller = {$in: [database.hash(controller)]};
+  }
+  const record = await database.collections.vcIssuerInstance.findOne(
+    query, {_id: 0, instance: 1});
+  if(!record) {
+    throw new BedrockError(
+      'Issuer instance not found.',
+      'NotFoundError',
+      {id, httpStatusCode: 404, public: true});
+  }
+  return record;
+};
+
+// get all issuer instances (optionally filtered by controller)
+exports.getAll = async ({controller} = {}) => {
+  const query = {};
+  if(controller) {
+    query.controller = database.hash(controller);
+  }
+  const records = await database.collections.vcIssuerInstance.find(
+    query, {_id: 0, instance: 1, meta: 1}).toArray();
+  return records;
+};
+
+// remove an issuer instance
+exports.remove = async ({id, controller}) => {
+  if(!(id && typeof id === 'string')) {
+    throw new TypeError('"id" must be a string.');
+  }
+  const query = {'instance.id': id};
+  if(controller) {
+    // only allow remove if controller matches
+    query.controller = {$in: [database.hash(controller)]};
+  }
+  const result = await database.collections.vcIssuerInstance.remove(query);
+  if(result.result.n === 0) {
+    throw new BedrockError(
+      'Issuer instance not found.',
+      'NotFoundError',
+      {id, httpStatusCode: 404, public: true});
+  }
+};
+
+// get a controller key for an instance
+exports.getControllerKey = async ({id}) => {
+  const record = await database.collections.vcIssuerInstance.findOne(
+    {'instance.id': id}, {_id: 0, controllerKeySeed: 1});
+  if(!record) {
+    throw new BedrockError(
+      'Issuer instance not found.',
+      'NotFoundError',
+      {id, httpStatusCode: 404, public: true});
+  }
+  const {controllerKeySeed: secret} = record;
+  const kmsClient = new KmsClient({
+    httpsAgent: new https.Agent({
+      // TODO: base on bedrock configuration
+      rejectUnauthorized: false
+    })
+  });
+  const controllerKey = await ControllerKey.fromSecret(
+    {secret, handle: id, kmsClient});
+  await ensureKeystore({controllerKey});
+  return controllerKey;
+};
+
+// set issuer and capabilities (zcaps) for an instance
+exports.setIssuer = async ({id, controller, issuer, zcaps}) => {
+  const query = {'instance.id': id};
+  if(controller) {
+    // only allow update if controller matches
+    query.controller = {$in: [database.hash(controller)]};
```

Comment:
Why is a new array being created here and `$in` being used?  Why not just direct compare?

Reply:
It should be checking `controller` for an array/string value and doing a map call if it's an array (or normalizing to an array). Filed #9.